### PR TITLE
wrf: add v4.2.2

### PR DESCRIPTION
**Pull request by stephenmsachs**

### CODE REVIEW

File: var/spack/repos/builtin/packages/wrf/package.py [hunk 1]
```diff
@@ -143,8 +144,8 @@ class Wrf(Package):
     patch("patches/4.0/tirpc_detect.patch", when="@4.0")
     patch("patches/4.0/add_aarch64.patch", when="@4.0")
 
-    patch("patches/4.2/arch.Config.pl.patch", when="@4.2:")
-    patch("patches/4.2/arch.configure.defaults.patch", when="@4.2")
+    patch("patches/4.2/arch.Config.pl.patch", when="@4.2:4.2.1,4.3:4.3.2")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/wrf/package.py:147]: Curious.  Why does the patch not specifically apply to `4.2.2` when it applies to `4.3:4.3.2`?

**Review by tldahlgren — COMMENTED** [review]
Confirmed sha256 but have a question about a patch.

File: var/spack/repos/builtin/packages/wrf/package.py (hunk 1 above)
Comment (stephenmsachs) [reply var/spack/repos/builtin/packages/wrf/package.py:147]: Yes, their release schedule is a little strange. The [fix](https://github.com/wrf-model/WRF/pull/1552) went in on Jan 6,2022 and made it into 4.2.2 (released Jan 15). Version 4.3.2 (release Dec 15, 2021) did not get the patch.

Comment (haampie) [reply var/spack/repos/builtin/packages/wrf/package.py:147]: It's just normal backports, right

**Comment by haampie** [comment]
This has conflicts now

**Comment by stephenmsachs** [comment]
Version has been added.

**Pull request closed without merging**